rockcraft/oci.py

```diff
@@ -313,8 +325,18 @@ def set_annotations(self, annotations: Dict[str, Any]) -> None:
         emit.progress(f"Labels and annotations set to {labels_list}", permanent=True)
 
 
-def _copy_image(source: str, destination: str, *system_params: str) -> None:
-    """Transfer images from source to destination."""
+def _copy_image(
+    source: str,
+    destination: str,
+    *system_params: str,
+    copy_params: Optional[List[str]] = None,
+) -> None:
+    """Transfer images from source to destination.
+
+    :param copy_params: Extra parameters to pass specifically to the "copy"
+        skopeo command (and not to skopeo itself).
+    """
+    copy_extra = copy_params if copy_params else []
```

Comment:
Would you be able to get rid of this line if you changed the parameter from
`copy_params: Optional[List[str]] = None` to `copy_params: List[str] = []`?

Reply:
Yes, but I thought our "standard" regarding optional parameters was to have `None` instead (I'm not sure on that one actually)

Reply:
Interesting, I hadn't thought about that.  Now I had to look it up:

Occurrences of `<param>: Optional[List...] = None` in snapcraft, charmcraft, and rockcraft = 14
Occurrences of `<param>: List[...] = []` in snapcraft, charmcraft, and rockcraft = 45

I'm fine with either choice.  If you have a preference, I will add it to my list of team standards, which is probably ready to be published somewhere for the whole team.

Reply:
the potential problems with mutables exist when assigning to a paramater. So None is the safe choice

- https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments